src/analysis/bounded_variation.lean

```diff
@@ -753,6 +791,50 @@ begin
         evariation_on.Icc_add_Icc f xy yz ys], },
 end
 
+@[protected]
+lemma edist_zero_of_eq_zero
+  {f : α → E} {s : set α} (hf : has_locally_bounded_variation_on f s)
+  {a b : α} (ha : a ∈ s) (hb : b ∈ s) (h : variation_on_from_to f s a b = 0) :
+  edist (f a) (f b) = 0 :=
+begin
+  apply le_antisymm _ (zero_le _),
+  rw ←ennreal.of_real_zero,
+  rcases le_total a b with h'|h',
+  work_on_goal 2
+  { rw [eq_neg_swap, neg_eq_zero] at h,
+    rw [edist_comm],
+    swap_var [a b, ha ↔ hb], },
+  all_goals
+  { rw [←h, eq_of_le f s h', ennreal.of_real_to_real (hf a b ha hb)],
+    apply evariation_on.edist_le,
+    exacts [⟨ha, ⟨le_rfl, h'⟩⟩, ⟨hb, ⟨h', le_rfl⟩⟩], },
+end
+
+@[protected]
+lemma eq_left_iff
+  {f : α → E} {s : set α} (hf : has_locally_bounded_variation_on f s)
+  {a b c : α} (ha : a ∈ s) (hb : b ∈ s) (hc : c ∈ s) :
+  variation_on_from_to f s a b = variation_on_from_to f s a c ↔ variation_on_from_to f s b c = 0 :=
+by simp only [←add hf ha hb hc, self_eq_add_right]
+
+@[protected]
+lemma eq_zero_iff_of_le
+  {f : α → E} {s : set α} (hf : has_locally_bounded_variation_on f s)
+  {a b : α} (ha : a ∈ s) (hb : b ∈ s) (ab : a ≤ b) :
+  variation_on_from_to f s a b = 0 ↔
+    ∀ ⦃x⦄ (hx : x ∈ s ∩ Icc a b) ⦃y⦄ (hy : y ∈ s ∩ Icc a b), edist (f x) (f y) = 0 :=
+by rw [eq_of_le _ _ ab, ennreal.to_real_eq_zero_iff,
+       or_iff_left (hf a b ha hb), evariation_on.eq_zero_iff]
+
+@[protected]
+lemma eq_zero_iff_of_ge
+  {f : α → E} {s : set α} (hf : has_locally_bounded_variation_on f s)
+  {a b : α} (ha : a ∈ s) (hb : b ∈ s) (ba : b ≤ a) :
+  variation_on_from_to f s a b = 0 ↔
+    ∀ ⦃x⦄ (hx : x ∈ s ∩ Icc b a) ⦃y⦄ (hy : y ∈ s ∩ Icc b a), edist (f x) (f y) = 0 :=
+by rw [eq_of_ge _ _ ba, neg_eq_zero, ennreal.to_real_eq_zero_iff,
+       or_iff_left (hf b a hb ha), evariation_on.eq_zero_iff]
```

Comment:
Maybe add a [set.uIcc](https://leanprover-community.github.io/mathlib_docs/data/set/intervals/unordered_interval.html#set.uIcc) version to unify both.